src/include/sof/audio/module_adapter/module/module_interface.h

```diff
@@ -143,4 +143,11 @@ struct module_interface {
 	int (*free)(struct processing_module *mod);
 };
 
+/* Convert first_block/last_block indicator to fragment position */
+static inline enum module_cfg_fragment_position
+first_last_block_to_frag_pos(bool first_block, bool last_block)
+{
+	return (last_block << 1) | first_block;
```

Comment:
This is nice and clever, I like it, but with the current definitions of `MODULE_CFG_FRAGMENT_*` it's a bit dangerous and error prone. To make this safe could you add to this patch a redefinition similar to
```
#define CFG_FRAGMENT_IS_LAST BIT(1)
#define CFG_FRAGMENT_IS_FIRST BIT(0)
#define MODULE_CFG_FRAGMENT_MIDDLE 0
#define MODULE_CFG_FRAGMENT_FIRST CFG_FRAGMENT_IS_FIRST
#define MODULE_CFG_FRAGMENT_LAST CFG_FRAGMENT_IS_LAST
#define MODULE_CFG_FRAGMENT_SINGLE (CFG_FRAGMENT_IS_FIRST | CFG_FRAGMENT_IS_LAST)
```

Reply:
@paulstelian97 @lyakh The suggestion does make it intuitive, and it is safer because the macros shadow the enum, which makes the enum unnecessary. 
With the macros defined, the enum is still used in some functions' signature, but the real thing we use is the macros we defined, this is confusing. You see, if people see a function return `enum module_cfg_fragment_position`, they will naturally think MODULE_CFG_FRAGMENT_FIRST is a member of the enum, but it is actually out macro.

This bitwise op does depends on the value order of the enum, but I don't see the possibility that it will be changed in the future.